File: ClothingStoreApp/ClothingStore.Application/DTOs/SizeDto.cs
namespace ClothingStore.Application.DTOs
{
    public record class SizeDto
    {
        public Guid SizeID { get; set; }
        public string Letter { get; set; } = string.Empty;
    }

    public record class ShirtSizeDto : SizeDto
    {
        public float Length { get; set; }
        public float ShoulderWidth { get; set; }
        public float ChestWidth { get; set; }
        public float SleeveLength { get; set; }
        public float SleeveCircumference { get; set; }
        public float NeckCircumference { get; set; }
    }

    public record class PantSizeDto: SizeDto
    {         
        public float Waist { get; set; }
        public float Inseam { get; set; }
        public float PantLegCircumference { get; set; }
    }

    public record class ShoeSizeDto: SizeDto
    {         
        public float Length { get; set; }
        public float Width { get; set; }
        public float? HeelHeight { get; set; }
    }

    public record class HatSizeDto: SizeDto
    {         
        public float Circumference { get; set; }
    }

    public record class DressSizeDto: SizeDto
    {         
        public float Bust { get; set; }
        public float Waist { get; set; }
        public float Hip { get; set; }
        public float Length { get; set; }
    }
}

File: ClothingStoreFrontend/src/Types/SizeDto.ts
export interface SizeDto{
    sizeId: string;
    letter: string;
}

export interface ShirtSizeDto extends SizeDto{
    length: number;
    shoulderWidth: number;
    chestWidth: number;
    sleeveLength: number;
    sleeveCircumference: number;
    neckCircumference: number;
}

export interface PantsSizeDto extends SizeDto{
    waist: number;
    inseam: number;
    pantLegCircumference: number;
}

export interface ShoeSizeDto extends SizeDto{
    length: number;
    eidth: number;
    heelHight?: number;
}

export interface HatSizeDto extends SizeDto{
    circumference: number;
}

export interface DressSizeDto extends SizeDto{
    bust: number;
    waist: number;
    hip: number;
    length: number;
}

export type AnySizeDto = ShirtSizeDto | PantsSizeDto | ShoeSizeDto | HatSizeDto | DressSizeDto;